### PR TITLE
feat(vanilla): add algolia-insights code sample

### PR DESCRIPTION
**Pull request by tkrugg**

### CODE REVIEW

File: InstantSearch.js/algolia-insights/index.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      </div>
+
+      <div class="right-panel">
+        <div id="searchbox" class="ais-SearchBox"></div>
+        <div id="hits"></div>
+        <div id="pagination"></div>
+      </div>
+    </div>
+
+    <script src="https://cdn.jsdelivr.net/npm/algoliasearch@3.32.0/dist/algoliasearchLite.min.js"></script>
```
Comment (francoischalifour) [inline InstantSearch.js/algolia-insights/index.html]:
```suggestion
    <script src="https://cdn.jsdelivr.net/npm/algoliasearch@3.32.1/dist/algoliasearchLite.min.js"></script>
```

File: InstantSearch.js/algolia-insights/index.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@3.0.0"></script>
```
Comment (francoischalifour) [inline InstantSearch.js/algolia-insights/index.html]:
```suggestion
    <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@3.2.1"></script>
```

File: InstantSearch.js/algolia-insights/src/app.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+document.addEventListener('click', e => {
+  if (e.target.matches('.button-click')) {
+    window.aa('clickedObjectIDsAfterSearch', {
+      index: search.indexName,
+      eventName: 'click-result',
+      objectIDs: [e.target.getAttribute('data-algolia-objectid')],
+      queryID: e.target.getAttribute('data-algolia-queryid'),
+      positions: [parseInt(e.target.getAttribute('data-algolia-position'))],
+      // parseInt because getAttribute always returns a string
```
Comment (francoischalifour) [inline InstantSearch.js/algolia-insights/src/app.js]: This comment should probably be before the above line.

File: InstantSearch.js/algolia-insights/src/app.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    });
+  } else if (e.target.matches('.button-conversion')) {
+    aa('convertedObjectIDsAfterSearch', {
+      index: search.indexName,
```
Comment (francoischalifour) [inline InstantSearch.js/algolia-insights/src/app.js]: I'm not sure we document the usage of `search[property]` anywhere. That's not officially supported.

Maybe we should rather store the `indexName` in a constant?

File: InstantSearch.js/algolia-insights/index.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    <script src="./src/app.js"></script>
+    <script>
+      !function(e,a,t,n,s,i,c){e.AlgoliaAnalyticsObject=s,e.aa=e.aa||function(){(e.aa.queue=e.aa.queue||[]).push(arguments)},i=a.createElement(t),c=a.getElementsByTagName(t)[0],i.async=1,i.src="https://cdn.jsdelivr.net/npm/search-insights@1.0.0",c.parentNode.insertBefore(i,c)}(window,document,"script",0,"aa");
```
Comment (francoischalifour) [inline InstantSearch.js/algolia-insights/index.html]:
```suggestion
      !function(e,a,t,n,s,i,c){e.AlgoliaAnalyticsObject=s,e.aa=e.aa||function(){(e.aa.queue=e.aa.queue||[]).push(arguments)},i=a.createElement(t),c=a.getElementsByTagName(t)[0],i.async=1,i.src="https://cdn.jsdelivr.net/npm/search-insights@1.0.1",c.parentNode.insertBefore(i,c)}(window,document,"script",0,"aa");
```

File: InstantSearch.js/algolia-insights/src/app.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    },
+    transformItems(items) {
+      const result = search.helper.lastResults;
+      const offset = result.hitsPerPage * result.page;
+
+      return items.map((item, index) => ({
+        ...item,
+        name: item.name.toUpperCase(),
+        queryID: result.queryID,
+        hitPosition: offset + index + 1,
```
Comment (Haroenv) [inline InstantSearch.js/algolia-insights/src/app.js]: I thought that was done by the change in the connector?

File: InstantSearch.js/algolia-insights/index.html (hunk 1 above)
Comment (tkrugg) [reply InstantSearch.js/algolia-insights/index.html]: I started off getting-started. 
if we do update them maybe let's update them all at the same time later for the sake of consistency?

File: InstantSearch.js/algolia-insights/src/app.js (hunk 4 above)
Comment (tkrugg) [reply InstantSearch.js/algolia-insights/src/app.js]: yes you're right

File: InstantSearch.js/algolia-insights/src/app.js (hunk 6 above)
Comment (tkrugg) [reply InstantSearch.js/algolia-insights/src/app.js]: When you say change you mean this PR https://github.com/algolia/instantsearch.js/pull/3598 ?
We'll do it in a later PR when we release this as a feature in IS.
For the moment, my goal is to have 1 insights example per flavour as the docs team expects it.

File: InstantSearch.js/algolia-insights/src/app.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+search.addWidget(
+  instantsearch.widgets.pagination({
+    container: '#pagination',
+  })
+);
+
+search.start();
+
+document.addEventListener('click', e => {
```
Comment (Haroenv) [inline InstantSearch.js/algolia-insights/src/app.js]:
```suggestion
document.getElementById('hits').addEventListener('click', e => {
```

Comment (francoischalifour) [reply InstantSearch.js/algolia-insights/src/app.js]: `document.querySelector('#hits')` might be more "recent".

**Comment by tkrugg** [comment]
@francoischalifour yes it does. As we talked about it the other day when you faced the same problem, there is some kind of bug on codesandbox with global variables that breaks all insights examples.

**Pull request merged**